common/functions/src/scalars/maths/sqrt.rs

```diff
@@ -14,64 +14,58 @@
 
 use std::fmt;
 
-use common_datavalues::prelude::*;
-use common_datavalues::DataTypeAndNullable;
-use common_exception::ErrorCode;
+use common_datavalues2::prelude::*;
+use common_datavalues2::with_match_primitive_type_id;
 use common_exception::Result;
+use num::cast::AsPrimitive;
 
-use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function2_factory::Function2Description;
+use crate::scalars::function_common::assert_numeric;
 use crate::scalars::function_factory::FunctionFeatures;
-use crate::scalars::Function;
+use crate::scalars::Function2;
+use crate::scalars::ScalarUnaryExpression;
 
 #[derive(Clone)]
 pub struct SqrtFunction {
     display_name: String,
 }
 
 impl SqrtFunction {
-    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function2>> {
         Ok(Box::new(SqrtFunction {
             display_name: display_name.to_string(),
         }))
     }
 
-    pub fn desc() -> FunctionDescription {
-        FunctionDescription::creator(Box::new(Self::try_create))
+    pub fn desc() -> Function2Description {
+        Function2Description::creator(Box::new(Self::try_create))
             .features(FunctionFeatures::default().deterministic().num_arguments(1))
     }
 }
 
-impl Function for SqrtFunction {
+fn sqrt<S>(value: S) -> f64
+where S: AsPrimitive<f64> {
+    value.as_().sqrt()
+}
+
+impl Function2 for SqrtFunction {
     fn name(&self) -> &str {
         &*self.display_name
     }
 
-    fn return_type(&self, args: &[DataTypeAndNullable]) -> Result<DataTypeAndNullable> {
-        let data_type = if args[0].is_numeric() || args[0].is_string() || args[0].is_null() {
-            Ok(DataType::Float64)
-        } else {
-            Err(ErrorCode::IllegalDataType(format!(
-                "Expected numeric types, but got {}",
-                args[0]
-            )))
-        }?;
-
-        // sqrt on negative returns null, so nullable should be true.
-        Ok(DataTypeAndNullable::create(&data_type, true))
```

Comment:
It looks like `ScalarUnaryExpression` cannot handle `Option<f64>` directly, for now I let sqrt on negative returns `f64::NAN`.

cc @sundy-li , should we continue to return `NULL`?

Reply:
> now I let sqrt on negative returns f64::NAN

It's better now.